plugins/generic/usageStats/UsageStatsPlugin.inc.php

```diff
@@ -306,9 +306,44 @@ function getUsageEventCurrentDayLogName() {
 	 * @param $usageEvent array
 	 */
 	function _writeUsageEventInLogFile($usageEvent) {
+		$salt = null;
 		if ($this->getSetting(CONTEXT_ID_NONE, 'dataPrivacyOption')) {
-			// TODO: Get the salt
-			$salt = 'ToBeSolved';
+			// Salt management.
+			if (!Config::getVar('usageStats', 'salt_filepath')) return false;
+			$saltFilename = Config::getVar('usageStats', 'salt_filepath');
+			$currentDate = date("Ymd");
+			$saltFilenameLastModified = date("Ymd", filemtime($saltFilename));
+			$file = fopen($saltFilename, 'r');
+			$salt = trim(fread($file,filesize($saltFilename)));
+			fclose($file);
+			if (empty($salt) || ($currentDate != $saltFilenameLastModified)) {
+				if(function_exists('mcrypt_create_iv')) {
+					$newSalt = bin2hex(mcrypt_create_iv(16, MCRYPT_DEV_URANDOM|MCRYPT_RAND));
+				} elseif (function_exists('openssl_random_pseudo_bytes')){
+					$newSalt = bin2hex(openssl_random_pseudo_bytes(16, $cstrong));
+				} elseif (file_exists('/dev/urandom')){
+					$newSalt = bin2hex(file_get_contents('/dev/urandom', false, null, 0, 16));
+				} else {
+					$newSalt = mt_rand();
+				}
+				$file = fopen($saltFilename,'wb');
+				if (flock($file, LOCK_EX)) {
+					fwrite($file, $newSalt);
+					flock($file, LOCK_UN);
+				} else {
+					ScheduledTask::addExecutionLogEntry(__('plugins.generic.usageStats.saltWriteError', array()), SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
+				}
+				fclose($file);
+				$salt = $newSalt;
+			}
+		}
+
+		// Manage the IP address (evtually hash it)
+		if ($this->getSetting(CONTEXT_ID_NONE, 'dataPrivacyOption')) {
+			if (!isset($salt)) {
+				ScheduledTask::addExecutionLogEntry(__('plugins.generic.usageStats.hashSaltMissing', array()), SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
```

Comment:
But UsageStatsPlugin doesn't extend ScheduledTask, right?  So `$this` won't have access to the method, I think.